test-app/tests/helpers/file-queue-helper-test.js

```diff
@@ -110,6 +113,90 @@ module('Integration | Helper | file-queue', function (hooks) {
     assert.dom('[data-test-file]').doesNotExist();
   });
 
+  test('will be notified when an upload starts', async function (assert) {
+    this.server.post(
+      '/upload-file',
+      uploadHandler(() => {})
+    );
+
+    this.upload = (file) => file.upload('/upload-file');
+
+    this.uploadStarted = (file) =>
+      assert.step(`upload started: ${file.name}, state: ${file.state}`);
```

Comment:
Test is easier to read if you split it up into two assertions in my opinion. One checking that the callback is executed. One checking that file state is as expected.

```suggestion
      assert.step('upload started');
      assert.propContains(file, { name: dingus.txt, state: uploading });
```

Reply:
Cool. Not used `propContains` before – thanks for the tip!

Reply:
Unfortunately it doesn't seem like `propContains` supports class properties. Have separated the assertions for you, though.